client/app/dbaas/logs/detail/inputs/logs-inputs.service.js

```diff
@@ -0,0 +1,265 @@
+            })
+            .catch(err => {
+                this._resetAllCache();
+                return this._handleError("logs_inputs_restart_error", err, { inputTitle: input.info.title });
+            });
+    }
+
+    /**
+     * start an input
+     *
+     * @param {any} serviceName
+     * @param {any} inputId
+     * @returns promise which will resolve with the operation object
+     * @memberof LogsInputsService
+     */
+    startInput (serviceName, input) {
+        return this.InputsApiLexiService.start({ serviceName, inputId: input.info.inputId })
+            .$promise
+            .then(operation => {
+                this._resetAllCache();
+                return this._handleSuccess(serviceName, operation.data, "logs_inputs_start_success", { inputTitle: input.info.title });
+            })
+            .catch(err => {
+                this._resetAllCache();
+                return this._handleError("logs_inputs_start_error", err, { inputTitle: input.info.title });
+            });
+    }
+
+    /**
+     * stop an input
+     *
+     * @param {any} serviceName
+     * @param {any} inputId
+     * @returns promise which will resolve with the operation object
+     * @memberof LogsInputsService
+     */
+    stopInput (serviceName, input) {
+        return this.InputsApiLexiService.end({ serviceName, inputId: input.info.inputId })
+            .$promise
+            .then(operation => {
+                this._resetAllCache();
+                return this._handleSuccess(serviceName, operation.data, "logs_inputs_stop_success", { inputTitle: input.info.title });
+            })
+            .catch(err => {
+                this._resetAllCache();
+                return this._handleError("logs_inputs_stop_error", err, { inputTitle: input.info.title });
+            });
+    }
+
+    /**
+     * transforms the input by adding some additional information
+     *
+     * @param {any} input
+     * @returns the transformed input
+     * @memberof LogsInputsService
+     */
+    transformInput (input) {
+        input.info.engine.software = [input.info.engine.name, input.info.engine.version].join(" ");
+        input.actionsMap = input.actions.reduce((actions, action) => {
+            actions[action.type] = action.isAllowed;
+            return actions;
+        }, {});
+
+        const isProcessing = input.info.status === this.LogsInputsConstant.status.PROCESSING;
+        const isToBeConfigured = input.info.status === this.LogsInputsConstant.status.INIT && !input.actionsMap.START;
+        const isPending = (input.info.status === this.LogsInputsConstant.status.INIT || input.info.status === this.LogsInputsConstant.status.PENDING) &&
+                          input.actionsMap.START;
+        const isRunning = input.info.status === this.LogsInputsConstant.status.RUNNING;
+
+        input.info.state = isProcessing ? this.LogsInputsConstant.state.PROCESSING :
```

Comment:
`
switch (input.info.status) {
case this.LogsInputsConstant.status.PROCESSING :
  return this.LogsInputsConstant.state.PROCESSING;
case this.LogsInputsConstant.status.INIT:
  if (!input.actionsMap.START)
    return this.LogsInputsConstant.state.TO_CONFIGURE;
  else 
    return this.LogsInputsConstant.state.PENDING;
case this.LogsInputsConstant.status.PENDING:
  return this.LogsInputsConstant.state.PENDING;
case this.LogsInputsConstant.status.RUNNING:
  return this.LogsInputsConstant.state.RUNNING;
case this.LogsInputsConstant.state.PROCESSING:
  return this.LogsInputsConstant.state.PROCESSING;
default : return this.LogsInputsConstant.state.UNKNOWN;
  
}
`

Reply:
I've made some mistakes but you get the idea

Reply:
We cannot directly have a switch case here, because, some state like require restart, needs only a boolean flag check